app/src/main/java/com/github/sdp_begreen/begreen/firebase/FirebaseDB.kt

```diff
@@ -337,4 +338,22 @@ object FirebaseDB {
         }
         return locations
     }
+
+    /**
+     * Retrieves the list of advices from the realtime database
+     *
+     * @return the set of all advices
+     */
+    suspend fun getAdvices(): Set<String> {
+
+        val childrens = databaseReference.child(ADVICES_LOCATION_PATH).get().await().children
+        val advicesList = mutableSetOf<String>()
+
+        childrens.forEach { postSnapshot ->
+            // If the advice is not of type String, ignore it (should never happen in theory)
+            val advice = postSnapshot.value as? String
+            advice?.also { advicesList.add(it) }
+        }
+        return advicesList

```

Comment:
Here I would suggest you to return an `immutable Set` instead of a `mutable` one. 

I suggest you to do it like this, what do you think of it ?

```suggestion
        childrens.mapNotNull {
            val advice = it.value as? String
            advice
        }.toSet()
```